File: todo-app/app/schemas/todo.py
from datetime import datetime
from enum import Enum

from pydantic import BaseModel, Field


class Priority(str, Enum):
    LOW = "low"
    MEDIUM = "medium"
    HIGH = "high"


class TodoBase(BaseModel):
    title: str = Field(..., min_length=1, max_length=255)
    description: str | None = None
    priority: Priority = Priority.MEDIUM
    due_date: datetime | None = None


class TodoCreate(TodoBase):
    pass


class TodoUpdate(BaseModel):
    title: str | None = Field(None, min_length=1, max_length=255)
    description: str | None = None
    priority: Priority | None = None
    due_date: datetime | None = None
    completed: bool | None = None


class Todo(TodoBase):
    id: int
    completed: bool
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}


class TodoList(BaseModel):
    todos: list[Todo]
    total: int

File: todo-app/todo-frontend/todo-frontend/src/types/todo.ts
// Todo-related types

export interface Todo {
  id: number;
  title: string;
  description?: string;
  completed: boolean;
  priority: 'low' | 'medium' | 'high';
  due_date?: string;
  created_at: string;
  updated_at: string;
}

export interface TodoCreate {
  title: string;
  description?: string;
  priority?: 'low' | 'medium' | 'high';
  due_date?: string;
  completed?: boolean;
}

export interface TodoUpdate {
  title?: string;
  description?: string;
  priority?: 'low' | 'medium' | 'high';
  due_date?: string;
  completed?: boolean;
}

export interface TodoListResponse {
  todos: Todo[];
  total: number;
}

export interface AgentResponse {
  response: string;
}

export interface AgentRequest {
  message: string;
}

export type Priority = 'low' | 'medium' | 'high';